File: q1_bases/p05/src/task2.py
def get_input() -> int | None:
    current = None
    while True:
        value: int = int(input('Enter value: '))
        if not value:
            break
        if current is None or value > current:
            current = value
    return current


def main() -> None:
    res: int | None = get_input()
    print(f"{res=}")


if __name__ == "__main__":
    main()

File: q2_algorithms/p12/src/task2.cpp
#include <iostream>

#define GET_INPUT(type, name)              \
    type name;                             \
    std::cout << "enter " << #name << ":"; \
    std::cin >> name

double get_sum(const double k) {
    const double value = 1 / k;
    return value == 1 ? value : value + get_sum(k - 1);
}

int main() {
    GET_INPUT(int, k);
    if (k <= 0) {
        throw std::invalid_argument("value must be greater than zero");
    }
    const auto res = get_sum(k);
    std::cout << "res: " << res << "\n";
    return 0;
}
